solutions/platinum/hr-SubtreesPaths.mdx

```diff
@@ -0,0 +1,191 @@
+---
+id: hr-SubtreesPaths
+source: HR
+title: Subtrees & Paths
+author: Rameez Parwez
+---
+
+[Official Editorial](https://www.hackerrank.com/challenges/subtrees-and-paths/editorial)
+
+## Intuition
+
+The problem requires us to perform two operations: adding a value $x$ to all nodes in the subtree rooted at $t$, and finding the maximum value on the path from $a$ to $b$. From the [Tutorial](https://usaco.guide/plat/hld?lang=cpp#tutorial) we know that Heavy-Light Decomposition(HLD) handles both operations efficiently. Therefore we will use HLD along with a Lazy Segment Tree for the query operations.
```

Comment:
I would still prefer a more detailed explanation. For example, how exactly are you going to use the segment tree to handle the operations?

Reply:
The explanation would be similar to that template explanation, that is why I didn't bother to write it again and mentioned the link below.

Reply:
I think you should at the very least explain how to perform subtree updates. Although it is just a simple range add on the preorder traversal, it's not specifically mentioned in the link.

Also, in general, reiterating the standard concepts is important for readers to understand the core ideas of HLD.

Reply:
added the explanation for this.